doc/AEPs/aep-0002.md

```diff
@@ -0,0 +1,200 @@
+the users to run `alr update` to obtain all dependencies. This is currently
+frowned upon, again because this can only work today for manifests that are
+platform-independent.
+
+Proposal
+--------
+
+The key change is to fully embrace manual edition of the manifest, and put it
+under version control, in a similar vein to other package managers like
+`cargo`.
+
+The only caveat concerns (at the moment) `alr with`, which is leveraged to
+comfortably make dependencies available in one step (and behind the scenes
+updating the manifest and the project file(s). Thus, there is a need to keep
+`alr with` functionality without regenerating the manifest file, to respect
+user's manual editions.  To meet both ends, the proposal is to add a new
+`[private]` top-level table, at the end of the manifest, properly identified
+for users as out-of-bounds.  Its purpose is to hold machine-generated changes
+that should be of no concern to users of Alire, just like the private part of
+the specification holds no interest for users of an Ada package.  This is
+detailed further in the [Implementation](#implementation) section.
+
+This `[private]` table could be leveraged in the future for other information
+besides that inserted by `alr with`. For example, the lockfile contents could
+be moved there to avoid exposing another file that is not user editable anyway.
+
+Advantages for packagers are:
+
+1. They can format/comment the manifest without their editions being lost.
+1. The whole publish process can be automated (see the
+[examples](#workflow-examples) section).
+
+Advantages for regular users:
+
+1. They can use `alr with` as currently, without manually editing the manifest
+(for now, for platform-independent crates).
+1. They can instruct their users to simply use `alr update`, `alr build`, or
+`alr run` to obtain dependencies, build, or run their projects, respectively.
+This is analogous to `cargo` usage.
+
+Why not `requirements.txt`
+--------------------------
+
+Another possibility would be to go pip's way and have something similar to its
+[requirements.txt](#references) file. This file contains a sequence of commands
+to pass to `pip install` to recreate the build environment for a project.
+
+The analogy for Alire would be to have a similar file with commands for `alr`.
+This has a few drawbacks with no clear advantage over `cargo`'s solution:
+
+1. It does not help packagers, unless we have complete manipulation of the 
+manifest file via `alr` commands.
+    1. This would be a huge undertaking, just to generate a file that is 
+easily maintained by hand.
+    1. We would have in practice a duplication of metadata; one copy in the
+manifest file and another copy in the requirements file, in two different 
+formats.
+    1. Same problems with manifest formatting/comments being lost.
+
+Implementation
+==============
+
+The proposal relies on the following changes:
+
+- `${crate}/alire/crate.toml` (manifest) is under version control and manually 
```

Comment:
I think we should move it to the root dir: `${crate}/alire.toml`.

Reply:
And make it always `alire` in name, I guess. I think it's less confusing. And the lock file in tow, too (on both naming and placement).